corehq/apps/locations/views.py

```diff
@@ -246,10 +247,16 @@ def mk_loctype(name, parent_type, administrative,
         for loc_type in loc_types:
             for prop in ['name', 'parent_type', 'administrative',
                          'shares_cases', 'view_descendants', 'pk']:
-                assert prop in loc_type, "Missing an organization level property!"
+                if prop not in loc_type:
+                    raise LocationConsistencyError("Missing an organization level property!")
             pk = loc_type['pk']
             if not _is_fake_pk(pk):
                 pks.append(loc_type['pk'])
+        if not (
+            len(loc_types) == len(set(lt['name'] for lt in loc_types))
+            and len(loc_types) == len(set(lt['code'] for lt in loc_types))
+        ):
+            raise LocationConsistencyError("'name' and 'code' are supposed to be unique")
```

Comment:
These errors just get handled as 500s yeah? since there is front-end validation that seems fine, we could update if it comes up for some race condition or something.